### PR TITLE
Free Trials: Empty the cart when a free trial item is added to it

### LINKED ISSUE #1922: Free trials: empty cart when containing a free trial
When the user adds a free trial to their cart we should empty it so that they don't try to purchase something else at the same time.

Connected to https://github.com/Automattic/wp-calypso/issues/286

### PR DESCRIPTION
**Pull request by Tug**
Fixes #1922 

#### Testing instructions
1. Checkout this branch `git checkout fix/1922-empty-cart-free-trial`.
2. Start from an empty cart and add a product different from a free trial to your cart.
3. Go back to the "Plans" page and click on one of the "Start Free Trial" button.
4. Check that your previous item has disappeared and that the Free Trial plan is the only item in your cart.

#### Reviews

- [x] Product Review
- [x] Code Review

### CODE REVIEW

File: client/lib/cart-values/cart-items.js [hunk 1]
```diff
@@ -81,6 +82,11 @@ function cartItemShouldReplaceCart( cartItem, cart ) {
 		return true;
 	}
 
+	if ( cartItem.free_trial ) {
+		// adding a free trial plan to your cart replaces the cart
+		return true;
+	}
+
```
Comment (scruffian) [inline client/lib/cart-values/cart-items.js]: We also need to consider the case when the user already has a free trial in their cart, and they add another product.

**Comment by Tug** [comment]
Thanks for reviewing @scruffian I have updated the PR.

File: client/lib/cart-values/cart-items.js [hunk 2]
```diff
@@ -81,6 +82,12 @@ function cartItemShouldReplaceCart( cartItem, cart ) {
 		return true;
 	}
 
+	if ( cartItem.free_trial || hasFreeTrial( cart ) ) {
```
Comment (scruffian) [inline client/lib/cart-values/cart-items.js]: We should use the `isFreeTrial` function here.

**Comment by scruffian** [comment]
Works as advertised

File: client/lib/cart-values/cart-items.js (hunk 2 above)
Comment (Tug) [inline client/lib/cart-values/cart-items.js]: You mean `hasOnlyFreeTrial` for the right part of the condition ?

**Comment by scruffian** [comment]
LGTM

**Pull request merged**